scipy/stats/tests/test_multivariate.py

```diff
@@ -1635,6 +1637,144 @@ def test_haar(self):
         res = kstest(x.ravel(), uniform(-np.pi, 2*np.pi).cdf)
         assert_(res.pvalue > 0.05)
 
+
+class TestMultivariateHypergeom(object):
+    def test_logpmf(self):
+        vals1 = multivariate_hypergeom.logpmf(x=[3, 4], m=[5, 10], n=7)
+        assert_allclose(vals1, -1.1198144963187122, rtol=1e-8)
+
+        vals2 = multivariate_hypergeom.logpmf(x=[3, 4], m=[5, 10], n=0)
+        assert_allclose(vals2, np.NAN, rtol=1e-8)
+
+        vals3 = multivariate_hypergeom.logpmf(x=[-3, 4], m=[5, 10], n=7)
+        assert_allclose(vals3, np.NINF, rtol=1e-8)
+
+        vals4 = multivariate_hypergeom.logpmf(x=[3, 4], m=[-5, 10], n=7)
+        assert_allclose(vals4, np.NAN, rtol=1e-8)
+
+    def test_reduces_hypergeom(self):
+        # test that the multivariate_hypergeom pmf reduces to the
+        # binomial pmf in the 2d case.
+        val1 = multivariate_hypergeom.pmf(x=[3, 1], m=[10, 5], n=4)
+        val2 = hypergeom.pmf(k=3, M=15, n=4, N=10)
+        assert_allclose(val1, val2, rtol=1e-8)
+
+        val1 = multivariate_hypergeom.pmf(x=[7, 3], m=[15, 10], n=10)
+        val2 = hypergeom.pmf(k=7, M=25, n=10, N=15)
+        assert_allclose(val1, val2, rtol=1e-8)
+
+    def test_rvs(self):
+        # test if `rvs` is unbiased and large sample size converges
+        # to the true mean.
+        rv = multivariate_hypergeom(m=[3, 5], n=4)
+        rvs = rv.rvs(size=1000, random_state=123)
+        assert_almost_equal(rvs.mean(0), rv.mean(), decimal=1)
+
+    def test_pmf(self):
+        vals0 = multivariate_hypergeom.pmf(x=[5], m=[5], n=5)
+        assert_allclose(vals0, 1, rtol=1e-8)
+
+        vals1 = multivariate_hypergeom.pmf(x=[3, 4], m=[5, 10], n=7)
+        assert_allclose(vals1, 0.3263403263403265, rtol=1e-8)
+
+        vals2 = multivariate_hypergeom.pmf(x=[[[3, 5],[0, 8]], [[-1, 9], [1, 1]]],
+                                           m=[5, 10], n=[[8, 8], [8, 2]])
+        assert_allclose(vals2, [[0.39160839, 0.00699301], [0, 0.47619048]], rtol=1e-6)
+
+        x = np.empty((0, 2), dtype=np.float64)
+        vals3 = multivariate_hypergeom.pmf(x=x, m=[4], n=0)
+        assert_equal(vals3, np.empty([], dtype=np.float64))
+
+        vals4 = multivariate_hypergeom.pmf([1, 2], [4, 5], 5)
+        assert_allclose(vals4, 0, rtol=1e-8)
+
+        vals5 = multivariate_hypergeom.pmf(x=[3, 3, 0], m=[5, 6, 7], n=6)
+        assert_allclose(vals5, 0.010773540185304875, rtol=1e-8)
+
+    def test_pmf_broadcasting(self):
+        vals0 = multivariate_hypergeom.pmf(x=[3, 4], m=[[5, 10], [10, 15]], n=7)
+        assert_allclose(vals0, [0.32634033, 0.34075307], rtol=1e-7)
+
+        vals1 = multivariate_hypergeom.pmf(x=[[1], [2]], m=[[3], [4]], n=[1, 3])
+        assert_allclose(vals1, [1., 0.], rtol=1e-8)
+
+        vals2 = multivariate_hypergeom.pmf(x=[[[1], [2]]], m=[[3], [4]], n=[1, 3])
+        assert_allclose(vals2, [[1., 0.]], rtol=1e-8)
```

Comment:
This latter assertion is somewhat difficult to distinguish from the former, suggest to add an inline comment to describe the difference to make it easy for future selves.

Reply:
This is testing broadcasting so these are dummy tests to ensure shapes are correct. I will add an inline comment for it!